src/workerd/api/sql.c++

```diff
@@ -167,8 +167,12 @@ auto SqlStorage::Cursor::iteratorImpl(jsg::Lock& js, jsg::Ref<Cursor>& obj, Func
   }
 
   auto& query = state.query;
+
+  obj->rowsRead = query.getRowsRead();
```

Comment:
Hmm, it seems wasteful to query these counters on every single row even if the app doesn't ever use them. (And even apps that do use them probably wouldn't check them on every row.)

Maybe we could have the implementation of `getRowsRead()` delegate to the underlying `Query` if it is still active. Then we only need to make a copy when the query is done.

Or maybe this isn't expensive enough to matter?

Reply:
They're cheap enough that this probably doesn't matter, but I agree that it looks wasteful, which is a trap for future programmers to fall into.

I took your suggestion of saving them off when the `Query` is done. I didn't see a way of invalidating a query out from under a `Cursor`, but if you know of one, then please tell me because that'll make the cursor falsely report 0 for row counts.